Source/Core/buildModuleUrl.js

```diff
@@ -142,6 +142,9 @@ buildModuleUrl.setBaseUrl = function (value) {
 
 /**
  * Gets the base URL for resolving modules.
+ *
+ * @function
+ * @returns {string} the configured base URL
```

Comment:
We do use capital `S` `String` throughout the codebase. Is there a reason that's changed in this file?

Reply:
This is one of those questions where I convinced myself of the answer years ago but forgot most of the justification in the meantime.  [Typescript is very opinionated about it](https://www.typescriptlang.org/docs/handbook/declaration-files/do-s-and-don-ts.html), but doesn't really offer much justification in their docs.  There's some good discussion on [this SO question](https://stackoverflow.com/questions/6793228/does-a-matter-whether-a-value-is-primitive-or-boxed).  (If you poke around on SO, you'll also find questions about "why do boxed types even exist?" with quotes from the language designers blaming management and generally regretting the decision.)

The short version is: lowercase-s `string` is a primitive and title-case `String` is an object ("box").  The most visible difference called out in that SO answer is that `typeof new String()` is `object`, not `string`. Primitives are boxed automatically if needed, but if only a primitive will do, you don't get free un-boxing.  I think the idea is that you probably almost never return `new String` in your code, but you might do a `typeof` check on a function argument at runtime, so the safest thing is to always use the primitive type in docs.

Reply:
I guess we could back it out?  I don't see the harm in making piecemeal changes, at least if we're touching adjacent docs (as I was in this case).  Not a big deal one way or another.

Reply:
Yes, thanks. I think we prefer the consistency. This is also how we've approach code style changes to the code base: All at once, ideally automating the change where possible.